src/encoded/static/components/image.js

```diff
@@ -5,177 +5,149 @@ import * as globals from './globals';
 
 
 // Fixed-position lightbox background and image
-class Lightbox extends React.Component {
-    constructor() {
-        super();
-
-        // Set initial React state.
-        this.state = { imgHeight: 0 };
-
-        // Bind this to non-React methods.
-        this.handleResize = this.handleResize.bind(this);
-    }
-
-    componentDidMount() {
-        globals.bindEvent(window, 'resize', this.handleResize);
-        this.setState({ imgHeight: this.lightbox.offsetHeight - 40 });
-    }
-
-    componentWillUnmount() {
-        globals.unbindEvent(window, 'resize', this.handleResize);
-    }
-
-    // Window resized; set max-height of image
-    handleResize() {
-        this.setState({ imgHeight: this.lightbox.offsetHeight - 40 });
-    }
-
-    render() {
-        const lightboxVisible = this.props.lightboxVisible;
-        const lightboxClass = `lightbox${lightboxVisible ? ' active' : ''}`;
-        const imgStyle = { maxHeight: this.state.imgHeight };
-
-        /* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */
-        return (
-            <div className={lightboxClass} onClick={this.props.clearLightbox} aria-label="Close" ref={(div) => { this.lightbox = div; }}>
-                <div className="lightbox-img">
-                    <a aria-label="Open image" data-bypass="true" href={this.props.lightboxImg}>
-                        <img src={this.props.lightboxImg} alt="Attachment from submitters" style={imgStyle} />
-                    </a>
-                    <button className="lightbox-close" aria-label="Close" onClick={this.clearLightbox} />
-                </div>
+const Lightbox = ({ lightboxVisible, clearLightbox, lightboxImg }) => {
+    // Height of the image within the light box.
+    const [imgHeight, setImgHeight] = React.useState(0);
+    const lightbox = React.useRef(null);
+
+    // Called when the window is resized to resize the image within it.
+    const handleResize = () => {
+        setImgHeight(lightbox.current.offsetHeight - 40);
+    };
+
+    // Called when a key is pressed to close the lightbox with ESC.
+    const handleKeyDown = (e) => {
+        if (e.keyCode === 27) {
+            clearLightbox();
+        }
+    };
+
+    React.useEffect(() => {
+        // Need to resize and attach event handlers at mount.
+        handleResize();
+        window.addEventListener('resize', handleResize);
```

Comment:
If not mistaken, React.useEffect is use tasking needed after mounting or unmounting like clean-up and unsubscring. If true, should event be wired up here?